tip/Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM python:3.9-slim-bullseye
+COPY tip.py /tip.py
+RUN pip install PyGithub --progress-bar off
+ENTRYPOINT ["/tip.py"]
```

Comment:
Why not `python /tip.py`?

Reply:
The script is executable (it has a shebang) because it can be used directly (see the composite action).
`ENTRYPOINT ["python", "/tip.py"]` is more verbose and it is functionally equivalent, isn't it?

Reply:
As I understand it, your form starts an intermediate Bash around Python, right?

```
bash
  python /tip.py
```

I would like to see it start Python directly without Bash.

-----
If the Python image sets ENTRYPOINT to `python`, then CMD could be used to specify parameters to entrypoint, right?

See https://docs.docker.com/engine/reference/builder/#entrypoint
> ENTRYPOINT has two forms:
>
> The exec form, which is the preferred form:
> `ENTRYPOINT ["executable", "param1", "param2"]`
>
> The shell form:
> `ENTRYPOINT command param1 param2`
>
> An ENTRYPOINT allows you to configure a container that will run as an executable.

See also https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact
> **Understand how CMD and ENTRYPOINT interact**
> Both CMD and ENTRYPOINT instructions define what command gets executed when running a container. There are few rules that describe their co-operation.
> 
> 1. Dockerfile should specify at least one of CMD or ENTRYPOINT commands.
> 2. ENTRYPOINT should be defined when using the container as an executable.
> 3. CMD should be used as a way of defining default arguments for an ENTRYPOINT command or for executing an ad-hoc command in a container.
> 4. CMD will be overridden when running the container with alternative arguments.

Reply:
> I would like to see it start Python directly without Bash.

As a rule of thumb, I prefer to always start a shell around Python, because that is the typically use case when we run a python script locally. The shell takes care of dealing with the environment, if needed.

However, I agree that it might be desirable to use CMD instead of ENTRYPOINT.